Celeste.Mod.mm/Mod/Everest/Everest.Loader.cs

```diff
@@ -600,31 +600,57 @@ public static class Loader {
 
                             gen = type.GetMethod(genName, new Type[] { typeof(Level), typeof(LevelData), typeof(Vector2), typeof(EntityData) });
                             if (gen != null && gen.IsStatic && gen.ReturnType.IsCompatible(typeof(Entity))) {
-                                loader = (level, levelData, offset, entityData) => (Entity) gen.Invoke(null, new object[] { level, levelData, offset, entityData });
+                                loader = (level, levelData, offset, entityData) => {
+                                    Entity e = (Entity) gen.Invoke(null, new object[] { level, levelData, offset, entityData });
+                                    (e as patch_Entity).EntityData = entityData;
+                                    return e;
+                                };
+                                // You cannot consistently determine the Type from the Method construction, so you can't map EntityDataName to Type from this method.
+// example: Load(Level level, LevelData levelData, Vector2 offset, EntityData entityData) => entityData.Bool("legacy") ? new ExampleClassOld(entityData, offset) : new ExampleClassNew(entityData, offset);
                                 goto RegisterEntityLoader;
                             }
 
                             ctor = type.GetConstructor(new Type[] { typeof(EntityData), typeof(Vector2), typeof(EntityID) });
                             if (ctor != null) {
-                                loader = (level, levelData, offset, entityData) => (Entity) ctor.Invoke(new object[] { entityData, offset, new EntityID(levelData.Name, entityData.ID + (patch_Level._isLoadingTriggers ? 10000000 : 0)) });
```

Comment:
You're losing the _isLoadingTriggers patch here.

Reply:
_isLoadingTriggers is now refactored to LevelData._isRegisteringTriggers 
We already make the assumption that LevelData.Triggers is always registering triggers, and now that EntityData contains its own ID per Data, we can assign the 10000000 offset to the ID in EntityData